### PR TITLE
56514 conditional rendering cerner styling

### PR DESCRIPTION
**Pull request by acaruso-oddball**
## Summary

Conditional rendering of the Header component has caused a bug in which AutoSSO does not properly work. Initially this rendering was necessary, as it ensured that the sign-in button did not render twice on the Unified Sign In Page. However, to get AutoSSO to work properly, the header needs to render regardless but display conditionally. In this pr, we've updated the CSS around the mega-menu header to allow for it to display on every page but the Unified Sign In Page. 

## Related issue(s)

[Update CSS around conditional rendering of the Header component. #57701](https://github.com/department-of-veterans-affairs/va.gov-team/issues/57701)

## Testing done

Unit testing

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |     Screenshot 2023-05-15 at 12 19 01 PM   |  Screenshot 2023-05-15 at 12 19 22 PM   |
| Desktop |   Screenshot 2023-05-15 at 12 18 07 PM    |  Screenshot 2023-05-15 at 12 01 53 PM   |

## What areas of the site does it impact?

The Header Component will conditionally render based on location of location in app. 

## Tasks
- [x] Update Header component to conditionally hide the container that contains the Sign In Modal using CSS, when user navigates to the Unified Sign-In Page. 
- [x] Clean up neighboring styling to accommodate such changes. 
- [x] Update tests to accommodate such change.

## Acceptance Criteria
- [x] Header component's visibility is conditional using CSS.  
- [x] All tests pass appropriately. 
### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Comment by acaruso-oddball** [comment]
Hello @department-of-veterans-affairs/va-platform-cop-frontend @department-of-veterans-affairs/vsp-identity @department-of-veterans-affairs/vsa-public-websites-frontend can I get a review on this pr please.

File: src/platform/site-wide/header/helpers/index.js [hunk 1]
```diff
@@ -29,6 +29,12 @@ export const showLegacyHeader = () => {
   if (!legacyHeader) {
     return;
   }
+  const signInVis = window.location.pathname === '/sign-in/';
+
+  if (signInVis) {
+    const line = document.querySelector('.menu-rule.usa-one-whole');
+    line.style.borderBottom = 'none';
+  }
```
Comment (asg5704) [inline src/platform/site-wide/header/helpers/index.js:37]: No. Let's use CSS in our SignInApp to update styles

File: src/platform/site-wide/mega-menu/components/MegaMenu.jsx [hunk 2]
```diff
@@ -18,6 +18,9 @@ export default class MegaMenu extends React.Component {
 
     this.mobileMediaQuery.addListener(this.resetDefaultState);
     document.body.addEventListener('click', this.handleDocumentClick, false);
+
+    this.signInVis =
+      window.location.pathname === '/sign-in/' ? 'hidden' : 'visibile';
```
Comment (asg5704) [inline src/platform/site-wide/mega-menu/components/MegaMenu.jsx:23]: No, remove it. Again lets use CSS in our SignInApp to update styles.

File: src/platform/site-wide/mega-menu/components/MegaMenu.jsx [hunk 3]
```diff
@@ -131,7 +134,11 @@ export default class MegaMenu extends React.Component {
     const hasOpenSubMenu = currentSection !== '';
 
     return (
-      <div className="login-container" {...display}>
+      <div
+        className="login-container"
+        {...display}
+        style={{ visibility: this.signInVis }}
+      >
```
Comment (asg5704) [inline src/platform/site-wide/mega-menu/components/MegaMenu.jsx:141]: Revert it.

File: src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx [hunk 4]
```diff
@@ -29,6 +29,8 @@ class SearchHelpSignIn extends Component {
     this.showHomepageCreateAccountBlock();
   }
 
+  signInVis = window.location.pathname === '/sign-in/' ? 'hidden' : 'visibile';
+
```
Comment (asg5704) [inline src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx:33]: No. Let's use CSS in our SignInApp to update styles

File: src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx [hunk 5]
```diff
@@ -120,6 +122,7 @@ class SearchHelpSignIn extends Component {
         className={`vads-u-display--flex vads-u-align-items--center${
           this.props.isHeaderV2 ? '' : ' vads-u-padding-top--1'
         }`}
+        style={{ visibility: this.signInVis }}
```
Comment (asg5704) [inline src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx:125]: Revert it.

**Review by asg5704 — CHANGES_REQUESTED** [review]
I noticed that there are 2 `/keepalive` requests occuring (when toggling the `liveKeepalive`) fix it so we only have 1.

File: src/applications/login/containers/SignInApp.jsx [hunk 6]
```diff
@@ -52,14 +52,23 @@ export function UnifiedSigninPage({ router, location }) {
     [isSiSEnabled],
   );
 
+  const css = `
+  #legacy-header > div:nth-child(3) > div.menu-rule.usa-one-whole {
+    display: none;
+  }
+`;
+
```
[the remark this answers is not in the document]
Comment (acaruso-oddball) [reply src/applications/login/containers/SignInApp.jsx]: updated

File: src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx (hunk 5 above)
Comment (acaruso-oddball) [reply src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx]: updated

File: src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx (hunk 4 above)
Comment (acaruso-oddball) [reply src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx]: updated

File: src/platform/site-wide/mega-menu/components/MegaMenu.jsx (hunk 3 above)
Comment (acaruso-oddball) [reply src/platform/site-wide/mega-menu/components/MegaMenu.jsx]: updated

File: src/platform/site-wide/mega-menu/components/MegaMenu.jsx (hunk 2 above)
Comment (acaruso-oddball) [reply src/platform/site-wide/mega-menu/components/MegaMenu.jsx]: updated

File: src/platform/site-wide/header/helpers/index.js (hunk 1 above)
Comment (acaruso-oddball) [reply src/platform/site-wide/header/helpers/index.js]: updated

File: src/applications/login/containers/SignInApp.jsx [hunk 7]
```diff
@@ -52,14 +51,35 @@ export function UnifiedSigninPage({ router, location }) {
...
   return (
     <>
-      <AutoSSO />
-      <LoginContainer
-        isUnifiedSignIn
-        externalApplication={externalApplication}
-        loggedOut={loggedOut}
-      />
+      <style>{css}</style>
+      <div>
```
Comment (asg5704) [inline src/applications/login/containers/SignInApp.jsx:76]: Did we need this extra `div`?

File: src/platform/site-wide/mega-menu/components/MegaMenu.jsx [hunk 8]
```diff
@@ -131,7 +131,7 @@ export default class MegaMenu extends React.Component {
     const hasOpenSubMenu = currentSection !== '';
 
     return (
-      <div className="login-container" {...display}>
+      <div className=" hidden-header login-container" {...display}>
```
Comment (asg5704) [inline src/platform/site-wide/mega-menu/components/MegaMenu.jsx:134]: Fix the spacing.

**Review by asg5704 — CHANGES_REQUESTED** [review]
Still requires some minor tweaks.

File: src/platform/site-wide/mega-menu/components/MegaMenu.jsx (hunk 8 above)
Comment (acaruso-oddball) [reply src/platform/site-wide/mega-menu/components/MegaMenu.jsx:134]: done

File: src/applications/login/containers/SignInApp.jsx [hunk 9]
```diff
@@ -52,14 +51,35 @@ export function UnifiedSigninPage({ router, location }) {
...
+  }
+
+  .menu-rule.usa-one-whole { 
+    visibility: hidden;
+  }
+
+  div .profile-nav-container {
+    visibility: hidden;
+  }
+
```
[the remark this answers is not in the document]
Comment (acaruso-oddball) [reply src/applications/login/containers/SignInApp.jsx:70]: updated

**Review by asg5704 — APPROVED** [review]

**Comment by acaruso-oddball** [comment]
Hello @department-of-veterans-affairs/vsa-public-websites-frontend @department-of-veterans-affairs/va-platform-cop-frontend can I get a review?

**Review by randimays — APPROVED** [review]
LGTM

**Pull request merged**